### DISCUSSION TITLE
Organization name or domain prompt page

### DISCUSSION THREAD

**Discussion by gabriel-hoff-vectorsolutions** [zitadel/zitadel#7676] [Q&A]
I have a use case where I'd like users to be prompted for their organization name (or primary domain) when applications start the authentication flow without specifying an organization scope (via `urn:zitadel:iam:org:domain:primary:{domainname}` or `urn:zitadel:iam:org:id:{id}`).

I believe Zitadel currently doesn't support anything like that, only identifying the user's organization via home realm discovery. Based on Zitadel's documentation, it seems that the customization level can be implemented; it's just not clear to me where to start from.

Could you guide me on possible approaches to add an organization prompt page before asking for the user's credentials?

### Context

We'd like to have the organization prompt page because:
1. Home realm discovery only works if users' email addresses are unique across organizations. If the same email address is used for users in different organizations, Zitadel is not able to log the user in.
2. Our users currently access some of our applications through Auth0 by presenting an organization prompt page. We'd like to keep that behavior with Zitadel.

I also accept any suggestions for other ways to solve this problem.

**Comment by hifabienne** [comment]
Hei @vs-gtadeu 
At the moment we do not provide an organization chooser.
And would recommend implementing this in your application. In that case you are able to send the scope of the organization afterwards to ZITADEL login.

**Reply by gabriel-hoff-vectorsolutions** [reply]
Hi @hifabienne. Unfortunately, that is not a feasible option for us. We have dozens of applications, so having each application build its organization prompt page is counterproductive.

At that moment, we are considering using https://github.com/zitadel/typescript to build that page, but having to control all UI login flows also complicates things.

Do you know if an organization prompt page has been discussed or considered before? Is that something your team would consider implementing at some point in the future?

**Reply by hifabienne** [reply]
Hei @vs-gtadeu 
As far as I remember we did not have that request before.
So at the moment we do not have that on our feature list also not for the future.
If we see that this is something that is requested by multiple customers we will for sure discuss this.
I recommend you adding a feature request with your use case, so we can add it to the backlog and collect if others would be interested in that as well.

**Reply by gabriel-hoff-vectorsolutions** [reply]
Sure, a feature request makes sense. Here you go: https://github.com/zitadel/zitadel/issues/7887
